buildchain/buildchain/docs.py

```diff
@@ -0,0 +1,98 @@
+# coding: utf-8
+
+"""Tasks to build the MetalK8s documentation and add it to the ISO.
+
+This module provides task to:
+- build the doc as HTML
+- build the doc as PDF
+- add the generated PDF into the ISO
+
+Overview;
+
+               ┌──────────┐
+           ───>│ doc:html │
+┌───────┐╱     └──────────┘
+│  doc  │
+└───────┘╲     ┌──────────┐
+           ───>│ doc:pdf  │
+               └──────────┘
+"""
+
+
+
+from collections import namedtuple
+from typing import Callable, Iterator
+from pathlib import Path
+
+from buildchain import config
+from buildchain import coreutils
+from buildchain import constants
+from buildchain import targets
+from buildchain import types
+from buildchain import utils
+
+
+DocTarget = namedtuple('DocTarget', ('name', 'command', 'target'))
+DOC_BUILD_DIR : Path = constants.ROOT/'docs/_build'
+DOC_PDF_FILE  : Path = DOC_BUILD_DIR/'latex/{}.pdf'.format(config.PROJECT_NAME)
+
+
+def task_documentation() -> types.TaskDict:
+    """Generate documentations for the ISO."""
+    return {
+        'actions': None,
+        'task_dep': [
+            '_doc_mkdir_root',
+            '_doc_deploy'
+        ],
+    }
+
+
+def task__doc_mkdir_root() -> types.TaskDict:
+    """Create the documentation root directory."""
+    return targets.Mkdir(
+        directory=constants.ISO_DOCS_ROOT, task_dep=['_iso_mkdir_root']
+    ).task
+
+
+def task__doc_deploy() -> types.TaskDict:
+    """Deploy the documentation on the ISO."""
+    source = DOC_PDF_FILE
+    target = constants.ISO_DOCS_ROOT/'user-guide.pdf'
+    return {
+        'title': utils.title_with_target1('COPY'),
+        'actions': [(coreutils.cp_file, (source, target))],
+        'targets': [target],
+        'task_dep': ['_doc_mkdir_root'],
+        'file_dep': [source],
+        'clean': True,
+    }
+
+
+def task_doc() -> Iterator[types.TaskDict]:
+    """Generate the documentation."""
+    def clean(target: DocTarget) -> Callable[[], None]:
+        """Delete the build sub-directory for the given target."""
+        return lambda: coreutils.rm_rf(target.target.parent)
+
+    doc_targets = (
```

Comment:
Fine for now, but could be useful to be able to choose, no? If developing docs, I guess `tox -e docs -- livehtml` is more comfortable, but `./doit.sh doc_html` could be cool as well IMHO

Reply:
I thought about supporting `livehtml`, but with `doit.sh auto doc:html` you'll have auto-recompilation as well no?

Reply:
That's not the only thing `livehtml` does: it recompiles, but also embeds some JS in the result such that when recompiling, the server can send an event to the browser to trigger a reload.

Reply:
Yeah, and it spawn a small webserver to serve the doc as well I think.

But meh, close enough :stuck_out_tongue:  (there is still `tox` if you really want to), or I could add a target `doc:live` (or `doc:auto`, or `doc:watch`), won't be hard.

I'll see once I have something working on the CI

Reply:
Done